### PR TITLE
chart: Add Snapshot Maximum Count to the Settings

### PR DESCRIPTION
**Pull request by yardenshoham**
It was missing

### CODE REVIEW

**Review by PhanLe1010 — COMMENTED** [review]
Thanks for the PR. There is a small naming issue.

Can we also verify the question.yaml is a valid yaml file or not by https://www.yamllint.com/? 
We will add this functionality in Longhorn in the future https://github.com/longhorn/longhorn/issues/7897

File: chart/templates/default-setting.yaml [hunk 1]
```diff
@@ -223,4 +223,7 @@ data:
     {{- end }}
     {{- if not (kindIs "invalid" .Values.defaultSettings.v2DataEngineGuaranteedInstanceManagerCPU) }}
     v2-data-engine-guaranteed-instance-manager-cpu: {{ .Values.defaultSettings.v2DataEngineGuaranteedInstanceManagerCPU }}
+    {{- end }}
+    {{- if not (kindIs "invalid" .Values.defaultSettings.maxSnapshotNum) }}
+    max-snapshotum: {{ .Values.defaultSettings.maxSnapshotNum }}
```
[the remark this answers is not in the document]
Comment (yardenshoham) [reply chart/templates/default-setting.yaml:228]: Oops I will fix

[the remark this answers is not in the document]
Comment (PhanLe1010) [reply chart/templates/default-setting.yaml]: Can you create a Github bug ticket @yardenshoham https://github.com/longhorn/longhorn/issues/new/choose

The ticket is used to move this fix through our development pipeline which including QA testing

[the remark this answers is not in the document]
Comment (yardenshoham) [reply chart/templates/default-setting.yaml]: Done https://github.com/longhorn/longhorn/issues/7906

[the remark this answers is not in the document]
Comment (innobead) [reply chart/templates/default-setting.yaml]: Please fix this to `snapshot-max-count` as Phan mentioned. Thanks.

File: chart/values.yaml [hunk 2]
```diff
@@ -319,6 +319,8 @@ defaultSettings:
   allowCollectingLonghornUsageMetrics: ~
   # -- Setting that temporarily prevents all attempts to purge volume snapshots.
   disableSnapshotPurge: ~
+  # -- Maximum snapshot count for a volume. The value should be between 2 to 250
+  maxSnapshotNum: ~
```
Comment (innobead) [inline chart/values.yaml]: `snapshotMaxCount` instead

**Review by innobead — CHANGES_REQUESTED** [review]

File: chart/README.md [hunk 3]
```diff
@@ -288,6 +288,7 @@ During installation, you can either allow Longhorn to use the default system set
 | defaultSettings.guaranteedInstanceManagerCPU | Percentage of the total allocatable CPU resources on each node to be reserved for each instance manager pod when the V1 Data Engine is enabled. The default value is "12". |
 | defaultSettings.kubernetesClusterAutoscalerEnabled | Setting that notifies Longhorn that the cluster is using the Kubernetes Cluster Autoscaler. |
 | defaultSettings.logLevel | Log levels that indicate the type and severity of logs in Longhorn Manager. The default value is "Info". (Options: "Panic", "Fatal", "Error", "Warn", "Info", "Debug", "Trace") |
+| defaultSettings.snapshotMaxCount | Maximum snapshot count for a volume. The value should be between 2 to 250 |
```
Comment (FrankYang0529) [inline chart/README.md:291]: Hi @yardenshoham, may you run `./scripts/helm-docs.sh` to update `README.md` file? Thanks.

Ref: https://github.com/longhorn/longhorn/wiki/Longhorn-Setting-Update

**Review by FrankYang0529 — CHANGES_REQUESTED** [review]
Overall LGTM. Thanks for the contribution.

File: chart/templates/default-setting.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yardenshoham) [reply chart/templates/default-setting.yaml]: Done https://github.com/longhorn/longhorn/pull/7896/commits/9ab25c13ecb67dac3590bf92d52522e19b35c1df

File: chart/values.yaml (hunk 2 above)
Comment (yardenshoham) [reply chart/values.yaml]: Done https://github.com/longhorn/longhorn/pull/7896/commits/9ab25c13ecb67dac3590bf92d52522e19b35c1df

File: chart/README.md (hunk 3 above)
Comment (yardenshoham) [reply chart/README.md]: Done https://github.com/longhorn/longhorn/pull/7896/commits/ba17bf90fe38f5409fd13068e15679a22c9af236

**Review by FrankYang0529 — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
Thanks @yardenshoham !

**Comment by innobead** [comment]
@mergify backport v1.6.x

**Pull request merged**